addons/l10n_fr_hr_work_entry_holidays/models/hr_contract.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import pytz
+
+from odoo import models
+from odoo.addons.resource.models.resource import datetime_to_string
+
+class HrContract(models.Model):
+    _inherit = 'hr.contract'
+
+    def _get_contract_work_entries_values(self, date_start, date_stop):
+        # Add the work entries difference for french payroll
+        # Work entries by default are not generated on days the employee does not work
+        # So we have to fill the gaps with work entries for those periods
```

Comment:
```python
contracts_fr = self.filtered(lambda x: x.company_id.country_id.code == 'FR')
result = super(HrContract, self - contracts_fr)._get_contract_work_entries_values(date_start, date_stop)
for contract in contracts_fr:
     result += ...
```

Reply:
This function is only meant to fill the gaps, we still have to get the result from super for all the rest of the work entries :)